scipy/optimize/__init__.py

```diff
@@ -198,71 +178,155 @@
 
 .. autosummary::
    :toctree: generated/
 
-   linprog_verbose_callback -- Sample callback function for linprog (simplex)
+   LbfgsInvHessProduct - Linear operator for L-BFGS approximate inverse Hessian.
 
-Assignment problems:
+Benchmark Problems
+------------------
 
 .. autosummary::
    :toctree: generated/
 
-   linear_sum_assignment -- Solves the linear-sum assignment problem
+   rosen - The Rosenbrock function.
+   rosen_der - The derivative of the Rosenbrock function.
+   rosen_hess - The Hessian matrix of the Rosenbrock function.
+   rosen_hess_prod - Product of the Rosenbrock Hessian with a vector.
 
-Utilities
-=========
+Legacy Functions
+================
+
+The functions below are not recommended for use in new scripts;
+all of these methods are accessible via a newer, more consistent
+interfaces, provided by the interfaces above.
+
+Optimization
+------------
+
+General-purpose multivariate methods:
+
+.. autosummary::
+   :toctree: generated/
+
+   fmin - Nelder-Mead Simplex algorithm.
+   fmin_powell - Powell's (modified) level set method.
+   fmin_cg - Non-linear (Polak-Ribiere) conjugate gradient algorithm.
+   fmin_bfgs - Quasi-Newton method (Broydon-Fletcher-Goldfarb-Shanno).
+   fmin_ncg - Line-search Newton Conjugate Gradient.
+
+Constrained multivariate methods:
+
+.. autosummary::
+   :toctree: generated/
+
+   fmin_l_bfgs_b - Zhu, Byrd, and Nocedal's constrained optimizer.
+   fmin_tnc - Truncated Newton code.
+   fmin_cobyla - Constrained optimization by linear approximation.
+   fmin_slsqp - Minimization using sequential least-squares programming.
+   differential_evolution - stochastic minimization using differential evolution.
+
+Univariate (scalar) minimization methods:
+
+.. autosummary::
+   :toctree: generated/
+
+   fminbound - Bounded minimization of a scalar function.
+   brent - 1-D function minimization using Brent method.
+   golden - 1-D function minimization using Golden Section method.
+
+Least-Squares
+-------------
+
+.. autosummary::
+   :toctree: generated/
+
+   leastsq - Minimize the sum of squares of M equations in N unknowns.
+
+Root Finding
+------------
+
+General nonlinear solvers:
 
 .. autosummary::
    :toctree: generated/
 
-   approx_fprime - Approximate the gradient of a scalar function
-   bracket - Bracket a minimum, given two starting points
-   check_grad - Check the supplied derivative using finite differences
-   line_search - Return a step that satisfies the strong Wolfe conditions
+   fsolve - Non-linear multi-variable equation solver.
+   broyden1 - Broyden's first method.
+   broyden2 - Broyden's second method.
+
+Large-scale nonlinear solvers:
+
+.. autosummary::
+   :toctree: generated/
+
+   newton_krylov
+   anderson
+
+Simple iteration solvers:
 
-   show_options - Show specific options optimization solvers
-   LbfgsInvHessProduct - Linear operator for L-BFGS approximate inverse Hessian
-   HessianUpdateStrategy - Interface for implementing Hessian update strategies
```

Comment:
It seems that the `HessianUpdateStrategy` line got dropped (unless I missed it). I guess it should stay along with `LbfgsInvHessProduct` in "Hessian Approximation" section above.

Reply:
Ok, just included this line back.